### ISSUE TITLE
Gather client & server logs in verbose mode in test_suite_ssl

### ISSUE THREAD

**Issue by AndrzejKurek** [Mbed-TLS/mbedtls#6210]
Currently there's no easy way to view server or client logs from a local connection used in `test_suite_ssl`. 
It would greatly improve debugging possibilities if logs were gathered in a simillar fashion to `ssl-opt.sh` tests.
Proposed solution:
- Use the `--verbose` flag that's already present in unit tests to turn this feature on;
- With this option on create two files per every handshake - `o_tls_[cli/srv]_<TESTNUM>_<TESTNAME>.log` is a maximal approach that I could come with. Save them in the `tests` directory.
- Optional - add a dump of handshake options at the start of each file, similar to parameters that are used for `ssl-client2` or `ssl-server2`.

**Comment by gilles-peskine-arm** [comment]
I don't think we should reuse the existing `-v` option. What you want to print to standard output, and what you want to have archived in a file, are different concerns.

We do have to be a little careful about the total size of stdout and of logs when you run a lot of tests. In the past we've had problems because we were logging too much (hundreds of GB) and exceeded our space quota on a CI system.

`ssl-opt.sh` always saves logs to a file, and deletes the files if the test case passes and the `-p` option is not passed. It would be nice to arrange something similar, both so that we don't have to remember the differences and because it's generally a good compromise between having logs available and keeping space usage reasonable.

How about using an environment variable `MBEDTLS_SSL_DEBUG_LOG_PREFIX` and saving to something like
```
snprintf("%s-%s-%s-%u.log",
         getenv("MBEDTLS_SSL_DEBUG_LOG_PREFIX"),
         test_suite_name,
         is_server ? "srv" : "cli",
         test_case_number)
```
Don't save if the env variable isn't set and delete the logs on a passing test. Have a command line option to preserve logs even on success and optionally increase the log level.

**Comment by AndrzejKurek** [comment]
I like the idea of always having this option on and preserving the logs only if a test failed / an additional flag was passed. 
The only thing I don't agree with is having the `test_suite_name` before the `test_case_number` - is `test_suite_name` `ssl`, or the particular test case name? If multiple tests fail it's easier to have them sorted by order of execution first.

**Comment by gilles-peskine-arm** [comment]
> The only thing I don't agree with is having the test_suite_name before the test_case_number - is test_suite_name ssl, or the particular test case name?

I was mimicking the naming scheme used by `compat.sh` and `ssl-opt.sh`: `(c|o)-(cli|srv)-NUMBER.log`. Unfortunately, individual test cases don't have names as such, they have descriptions which can contain characters that are not valid in file names (including `/`).

> If multiple tests fail it's easier to have them sorted by order of execution first.

I don't understand where you're going there. They will be sorted by order of execution (maybe easier to do with `%05u` than `%u` but the sh scripts use `%u`).

**Comment by AndrzejKurek** [comment]
Right, all good. I was worried you want to have the description placed before the number, and that would be bizzarre. 
We can still try to strip invalid characters and have the description in the file name - I think that this would also be helpful.
If not there - as the first line in the log.
